lib/IRGen/IRGen.cpp

```diff
@@ -490,10 +490,13 @@ static createTargetMachine(IRGenOptions &Opts, ASTContext &Ctx) {
   }
 
   // Create a target machine.
-  llvm::TargetMachine *TargetMachine
-    = Target->createTargetMachine(Triple.str(), CPU,
-                                  targetFeatures, TargetOpts, Reloc::PIC_,
-                                  CodeModel::Default, OptLevel);
+  auto cmodel = CodeModel::Default;
+  if (Triple.isWindowsCygwinEnvironment())
+    cmodel = CodeModel::Large;
```

Comment:
Hmm, doesnt this problem exist beyond just cygwin?  Perhaps it should be a 64-bit windows check here instead?